packages/augur-tools/src/flash/replay-logs-v1.ts

```diff
@@ -0,0 +1,184 @@
+      if (this.availableAccounts.length > 0) {
+        this.accounts[address] = this.availableAccounts.pop();
+      } else { // Create and fund new account.
+        const wallet = ethers.Wallet.createRandom();
+        this.accounts[address] = {
+          secretKey: wallet.privateKey,
+          publicKey: wallet.address,
+          balance: _1_HUNDRED_ETH,
+        };
+
+        const piggybankWallet = new ethers.Wallet(this.piggybank.secretKey, this.provider);
+        await piggybankWallet.sendTransaction({
+          to: wallet.address,
+          value: `0x${new BigNumber(_1_HUNDRED_ETH).toString(16)}`,
+        });
+      }
+    }
+
+    return this.accounts[address];
+  }
+
+  async Replay(logs: ParsedLog[]) {
+    for (const log of logs) {
+      await this.ReplayLog(log);
+    }
+  }
+
+  async ReplayLog(log: ParsedLog) {
+    switch(log.name) {
+      case "UniverseCreated": return this.UniverseCreated(log);
+      case "MarketCreated": return this.MarketCreated(log);
+      case "OrderCreated": return this.OrderCreated(log);
+      default:
+    }
+  }
+
+  async UniverseCreated(log: ParsedLog) {
+    const { parentUniverse, childUniverse, payoutNumberators } = log;
+
+    console.log(`Replaying UniverseCreated "${childUniverse}"`);
+
+    if (parentUniverse === NULL_ADDRESS) { // Deployment already gave us a genesis universe so just record it.
+      const user = await this.User(this.piggybank); // Piggybank is typically the Augur deployer, including genesis universe.
+      this.universes[childUniverse] = user.augur.addresses.Universe;
+    } else {
+      // TODO No need to support forking yet. But when it is supported, remember
+      //      that parentPayoutDistributionHash is part of the Universe contract.
+    }
+  }
+
+  async MarketCreated(log: ParsedLog) {
+    const {
+      universe, extraInfo, market, marketCreator, designatedReporter,
+      prices, marketType, numTicks, outcomes } = log;
+
+    console.log(`Replaying MarketCreated "${market}"`);
+    console.log(log);
+
+    const timestamp = `${today.getTime() / 1000}`;
+    const endTime = `${inOneMonths.getTime() / 1000}`;
+    const feeDivisor = new BigNumber(0);
+
+    const feePerCashInAttoCash = new BigNumber(10).pow(16); // 1% fee
+    const [ start, end ] = [ makeDate(timestamp), makeDate(endTime) ];
+    const marketLength = end.getTime() - start.getTime();
+    const adjustedEndTime = new BigNumber(Math.floor((Date.now() + marketLength) / 1000));
+
+    const user = await this.Account(marketCreator).then((account) => this.User(account));
+    const reporter = await this.Account(designatedReporter);
+
+    let result;
+    switch(marketType) {
+      case 0: // YES_NO
+        result = await user.createYesNoMarket({
+          endTime: adjustedEndTime,
+          feePerCashInAttoCash,
+          affiliateFeeDivisor: feeDivisor,
+          designatedReporter: reporter.publicKey,
+          extraInfo,
```

Comment:
category has moved to extraInfo. 

If I'm understanding this, V1 log has _topic on market creation and to be compliant with V2 _topic will need to be added to `categories` in extraInfo.

categories: [_topic, "from V1", ""]

_topic or category, don't rem what comes in on V1 log

Reply:
looks like it's just `topic`. added!